features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Shared/ApiKeyRestriction.jsx

```diff
@@ -0,0 +1,335 @@
+    ipList,
+    updateIpList,
+    restrictSchema,
+    updateRestrictSchema,
+    newReferer,
+    updateNewReferer,
+    refererList,
+    updateRefererList,
+  } = props;
+
+  const onRefererTextUpdate = (e) => {
+    updateNewReferer(e.target.value.trim());
+    if (e.target.value.trim() == "") {
+      setInvalidReferer(false);
+    }
+  };
+
+  const addRefererItem = () => {
+    if (newReferer != null && newReferer != "") {
+      setInvalidReferer(false);
+      refererList.push(newReferer.trim());
+      updateRefererList(refererList);
+      updateNewReferer("");
+    } else {
+      setInvalidReferer(true);
+    }
+  };
+
+  const deleteRefererItem = (refererItem) => {
+    refererList.splice(refererList.indexOf(refererItem), 1);
+    updateRefererList(refererList);
+  };
+
+  const onIpTextUpdate = (e) => {
+    updateNewIp(e.target.value.trim());
+    if (e.target.value.trim() == "") {
+      setInvalidIP(false);
+    }
+  };
+
+  const addIpItem = () => {
+    if (newIP != null && newIP != "") {
+      const ipv4RegExp = new RegExp(
+        [
+          "^((25[0-5]|2[0-4][0-9]|[01]?",
+          "[0-9][0-9]?).){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)",
+          "(/(3[0-2]|2[0-9]|1[0-9]|[0-9]))?$",
+        ].join("")
+      );
+
+      const ipv6RegExp = new RegExp(
+        [
+          "^(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]",
+          "{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:",
+          "[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4})",
+          "{1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|",
+          "([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]",
+          "{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:",
+          "[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:",
+          "(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}",
+          ":){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]).){3,3}(25[0-5]",
+          "|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:",
+          "((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]).){3,3}(25[0-5]|",
+          "(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))(/((1(1[0-9]|2[0-8]))",
+          "|([0-9][0-9])|([0-9])))?$",
+        ].join("")
+      );
+
+      const validIP = ipv4RegExp.test(newIP) || ipv6RegExp.test(newIP);
+      setInvalidIP(!validIP);
+      if (validIP) {
+        ipList.push(newIP);
+        updateIpList(ipList);
+        updateNewIp("");
+      }
+    }
+  };
+
+  const deleteIpItem = (ipItem) => {
+    ipList.splice(ipList.indexOf(ipItem), 1);
+    updateIpList(ipList);
+  };
+
+  const onRestrictSchemaChange = (e) => {
+    updateRestrictSchema(e.target.value);
+    updateIpList([]);
+    updateRefererList([]);
+    updateNewIp("");
+    updateNewReferer("");
+    setInvalidIP(false);
+    setInvalidReferer(false);
+  };
+
+  return (
+    <React.Fragment>
+      <Box
+        border={1}
+        borderRadius={5}
+        style={{ margin: "10px" , padding: "10px"}}
```

Comment:
Don't use inline styling. Let's define styles at the top.

Reply:
fixed